app/controllers/products_controller.rb

```diff
@@ -29,7 +29,7 @@ def show
       render_404_page and return unless @product
     else
       # Anyone can view timestamped products, but only certain people can view live [retired] products
-      authorize @product
+      return render "/products/retired" unless Pundit.policy(current_user, @product).show?
```

Comment:
I believe you'll have the `policy` helper available, and can omit `current_user` as an implied argument...
```suggestion
      return render "/products/retired" unless policy(@product).show?
```